extensions/amp-story/1.0/amp-story.js

```diff
@@ -986,16 +986,30 @@ export class AmpStory extends AMP.BaseElement {
 
         if (shouldReOpenAttachmentForPageId === this.activePage_.element.id) {
           const attachmentEl = this.activePage_.element.querySelector(
-            'amp-story-page-attachment, amp-story-page-outlink'
+            'amp-story-shopping-attachment, amp-story-page-attachment, amp-story-page-outlink'
           );
 
-          if (attachmentEl) {
+          if (
+            attachmentEl &&
+            attachmentEl.tagName !== 'AMP-STORY-SHOPPING-ATTACHMENT'
+          ) {
```

Comment:
I'm a bit confused about these lines now, do you need to add the `amp-story-shopping-attachment` selector if you're going to specifically write a `if` statement to remove it right after?

Reply:
Yeah, since we moved the check to initialize the history service if it is not null regardless if there is a page attachment, we don't need to query for `amp-story-shopping-attachment anymore.` removed.